### PR TITLE
Update initialJoins to use Client.Join

### PR DESCRIPTION
**Pull request by brattonross**

### CODE REVIEW

**Comment by brattonross** [comment]
Just realised that this breaks the test TestCanJoinChannel, I'll look into it

**Comment by pajlada** [comment]
The JOIN gets lost in the wind because it initially tries to JOIN when there's no connection.

https://github.com/gempir/go-twitch-irc/blob/master/client.go#L528 Here we check if the connection is active before we try to send the message.
If the connection is not active, that means we don't send the created "JOIN #gempiR" message from the test. However, regardless of the connection state, we still iterate through all channels in the "joined" list and remove them https://github.com/gempir/go-twitch-irc/blob/master/client.go#L533-L538

A way to solve this is to save the "connActive" state, and apply that to the "channels" map here: https://github.com/gempir/go-twitch-irc/blob/master/client.go#L534

**Comment by brattonross** [comment]
Aha! Great thinking, I was getting myself tied up in thinking about all sorts of issues with concurrency but this is an elegant solution

File: client.go [hunk 1]
```diff
@@ -523,15 +523,17 @@ func (c *Client) Join(channels ...string) {
 
 	// If we have an active connection, explicitly join
 	// before we add the joined channels to our map
+	connActive := c.connActive.get()
```
Comment (gempir) [inline client.go]: not an expert, but doesn't this kinda defeat the purpose of the atomic bool?

Comment (brattonross) [reply client.go]: Should we be checking if the connection is active for each join message and each channel in the map then?

Comment (gempir) [reply client.go]: hmm no nevermind. This might be fine here

Comment (pajlada) [reply client.go]: You are right @gempir, it does defeat the purpose of the atomic bool. There is a chance that inbetween the connActive check, the JOIN succeeds and initialJoins has been called.

**Comment by pajlada** [comment]
I made a small "lifecycle" of our connections (read from left to right)
https://raw.githubusercontent.com/gempir/go-twitch-irc/pajlada/document-connection-cycle/docs/join.png
It doesn't seem like the atomic behaviour we perform is actually an issue, because from initialJoins we can guarantee the state of the "connActive" variable. Might be a different matter if Join is called from outside of initialJoins however.

**Comment by gempir** [comment]
I think we should keep checking if the connection is each join as we do now. So this PR requires some changes

**Pull request merged**